analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Conversion.cs

```diff
@@ -18,18 +18,40 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors;
 
-internal sealed class Conversion : SimpleProcessor<IConversionOperationWrapper>
+internal sealed class Conversion : MultiProcessor<IConversionOperationWrapper>
 {
     protected override IConversionOperationWrapper Convert(IOperation operation) =>
         operation.ToConversion();
 
-    protected override ProgramState Process(SymbolicContext context, IConversionOperationWrapper conversion) =>
-        context.State[conversion.Operand] is { } value && IsBuildIn(conversion.OperatorMethod) // Built-in conversions only
-            ? context.SetOperationValue(value)
-            : context.State;
+    protected override ProgramState[] Process(SymbolicContext context, IConversionOperationWrapper conversion) =>
+        IsNotBuildIn(conversion.OperatorMethod) ? context.State.ToArray() : StateForBuildInConversion(context, conversion);
+
+    private static bool IsNotBuildIn(ISymbol symbol) =>
+        symbol?.ContainingType.IsAny(KnownType.PointerTypes) is false;
+
+    private static ProgramState[] StateForBuildInConversion(SymbolicContext context, IConversionOperationWrapper conversion) =>
+        IsNotUncertainTryCast(conversion)
+            ? StateForRegularConversion(context, context.State[conversion.Operand])
+            : StateForUncertainTryCast(context, context.State[conversion.Operand]);
+
+    private static bool IsNotUncertainTryCast(IConversionOperationWrapper conversion) =>
+        !conversion.IsTryCast
+        || conversion.Operand.Type.DerivesOrImplements(conversion.Type)
+        || (conversion.Operand.Type.IsNonNullableValueType() && conversion.Type.IsNullableValueType());
+
+    private static ProgramState[] StateForRegularConversion(SymbolicContext context, SymbolicValue operandValue) =>
```

Comment:
Let's do the `.ToArray()` on invocation side. It will simplify the syntax clutter here. And "State" name will be correct
```suggestion
    private static ProgramState RegularConversionState(SymbolicContext context, SymbolicValue operandValue) =>
```